<!--REQUEST_ID=5eecae95-984b-447c-9af8-7ab1b9446c32-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46186&attempt_id=69d45902-6a0d-48a9-9f5e-3adbefe4c0f5)

Attempted: python27Packages.ncclient

The following builds were skipped because they don't evaluate on x86_64-darwin: python36Packages.ncclient

<details><summary>Partial log (click to expand)</summary><p>

```
    self.testcase.assertLessEqual(total_time, self.upper * (1.0 + TOLERANCE))
AssertionError: 0.015455961227416992 not less than or equal to 0.015

----------------------------------------------------------------------
Ran 196 tests in 3.865s

FAILED (SKIP=99, failures=11)
builder for '/nix/store/zsy3603w0isff9xxx7kz2cp5am59l7k3-python2.7-selectors2-2.0.1.drv' failed with exit code 1
cannot build derivation '/nix/store/2k2628ipx1b62m6hq9h3cb5sjcncrycz-python2.7-ncclient-0.6.2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/2k2628ipx1b62m6hq9h3cb5sjcncrycz-python2.7-ncclient-0.6.2.drv' failed
```
</p></details>

--- a/pkgs/development/python-modules/ncclient/default.nix
+++ b/pkgs/development/python-modules/ncclient/default.nix
@@ -2,6 +2,7 @@
 , buildPythonPackage
 , fetchPypi
 , paramiko
+, selectors2
 , lxml
 , libxml2
 , libxslt
@@ -21,7 +22,7 @@ buildPythonPackage rec {
   checkInputs = [ nose rednose ];
 
   propagatedBuildInputs = [
-    paramiko lxml libxml2 libxslt
+    paramiko lxml libxml2 libxslt selectors2
   ];
 
   checkPhase = ''

--- a/pkgs/development/python-modules/selectors2/default.nix
+++ b/pkgs/development/python-modules/selectors2/default.nix
@@ -1,0 +1,29 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, nose, psutil, mock }:
+
+buildPythonPackage rec {
+  version = "2.0.1";
+  pname = "selectors2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "81b77c4c6f607248b1d6bbdb5935403fef294b224b842a830bbfabb400c81884";
+  };
+
+  checkInputs = [ nose psutil mock ];
+
+  checkPhase = ''
+    # https://github.com/NixOS/nixpkgs/pull/46186#issuecomment-419450064
+    # Trick to disable certain tests that depend on timing which
+    # will always fail on hydra
+    export TRAVIS=""
+    nosetests tests/test_selectors2.py
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://www.github.com/SethMichaelLarson/selectors2;
+    description = "Back-ported, durable, and portable selectors";
+    license = licenses.mit;
+    maintainers = [ maintainers.costrouc ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -536,6 +536,8 @@ in {
 
   seekpath = callPackage ../development/python-modules/seekpath { };
 
+  selectors2 = callPackage ../development/python-modules/selectors2 { };
+
   serversyncstorage = callPackage ../development/python-modules/serversyncstorage {};
 
   simpleeval = callPackage ../development/python-modules/simpleeval { };

